src/pages/Request.vue

```diff
@@ -27,11 +28,42 @@
 <script>
     import Config from '../config.js';
     import Store from '../store.js';
+
+    import mixin from '../mixins/mixins';
+    import constants from "../constants";
+
     export default {
         name: 'request',
+        mixins: [
+            mixin.messages
+        ],
         data: function () {
             return {
-                username: null
+                username: null,
+                netMgr: Store.netMgr
+            }
+        },
+        methods: {
+            isEmailValid: function (email) {
+                var re = /\S+@\S+/;
+                return re.test(email);
+            },
+            onRequestResetEmail: function () {
+                if (this.isEmailValid(this.username)) {
+
+                    return this.netMgr.apiPost('user/lost_password', {'email' : this.username},
+
+                        function (response) {
+                            this.setMessage(response.data[1], constants.MESSAGE_SUCCESS)
+                        }.bind(this),
+                        function (error) {
+                            // toString added as some strings returned enclosed in brackets.
+                            this.setMessage(error.response.data.email.toString(), constants.MESSAGE_ERROR);
```

Comment:
Invalid email address errors (user entered junk) that get through to the API are returned as a single member array. No idea why, hence we'll card this for a long grass.